### ISSUE TITLE
Models in "5.2 Edge Classification/Regression" do not use edge features for edge classification tasks, or the tutorials is unclear about it

### ISSUE THREAD

**Issue by MrSeemsGood** [dmlc/dgl#7154]
## 📚 Documentation

Hi all, I have a question about the "5.2 Edge Classification/Regression" tutorial.

When creating a test graph, there is this line:
```python
edge_pred_graph.edata['feature'] = torch.randn(1000, 10)
```
Which *supposes* that edge features of the graph will be used for creating the model.

But examining the tutorial further, it seems as if the network does not actually use these edge features at all, but rather node features only? I'm pretty new to DGL, so I was wondering if these edge features are still somehow used by graph networks under the hood. If not, this tutorial seems a bit misleading to me, especially since using edge features for edge label prediction is the exact task I'm looking for a GNN for. 

If possible, I'd greatly appreciate some examples on networks that can use both edge and node features to predict edge labels, because so far I haven't found any.

**Comment by jermainewang** [comment]
You are right. The synthetic edge features are not used in the later modeling. The complexity here is that there are multiple ways to incorporate edge features. For example, one option is to change the `MLPPredictor` as followings:

```python
class MLPPredictor(nn.Module):
    def __init__(self, in_features, out_classes):
        super().__init__()
        self.W = nn.Linear(in_features * 2, out_classes)

    def apply_edges(self, edges):
        h_u = edges.src['h']
        h_v = edges.dst['h']
        h_e = edges.data['h']
        score = self.W(torch.cat([h_u, h_v, h_e], 1))  # Concatenate edge and node embeddings
        return {'score': score}

    def forward(self, graph, h_node, h_edge):
        with graph.local_scope():
            graph.ndata['h'] = h_node
            graph.edata['h'] = h_edge   # Attach edge embeddings to graph.
            graph.apply_edges(self.apply_edges)
            return graph.edata['score']
```

One could also make similar modification to DotPredictor or message passing. I agree that we should demonstrate at least one option. Putting this as an item to fix.

**Comment by MrSeemsGood** [comment]
Sweet, thanks for the response! FYI, the way I incorporated edge features into the model is just by using `u_dot_e` in the `DotProductPredictor` and it worked just fine! 👍🏻
